arangod/VocBase/vocbase.cpp

```diff
@@ -2022,6 +2096,94 @@ TRI_voc_rid_t TRI_StringToRid(char const* p, size_t len, bool& isOld, bool warn)
   return HybridLogicalClock::decodeTimeStamp(p, len);
 }
 
+
+VocbaseOptions arangodb::getVocbaseOptions(VPackSlice const& options) {
+  // Invalid options will be silently ignored. Default values will be used
+  // instead.
+  //
+  // This Function will be called twice - the second time we do not run in
+  // the risk of consulting the ClusterFeature, because defaults are provided
+  // during the first call.
+
+  VocbaseOptions vocbaseOptions;
+  vocbaseOptions.replicationFactor = 1;
```

Comment:
I think here we should populate things with the system defaults, e.g. ClusterFeature::defaultReplicationFactor() etc.

Reply:
So we want to access the features even though the information could already be contained in the slice we are inspecting?